packages/clay-autocomplete/src/ClayAutocomplete.soy

```diff
@@ -0,0 +1,206 @@
+	]}
+	{@param? requestPolling: number}
+	{@param? requestRetries: number}
+	{@param? requestTimeout: number}
+	{@param? unstable_content: html}
+
+	{let $attributes kind="attributes"}
+		class="dropdown-full
+			{if $elementClasses}
+				{sp}{$elementClasses}
+			{else}
+				{sp}form-group
+			{/if}
+		"
+
+		data-onkeydown="{$_handleOnKeydown}"
+
+		{if $id}
+			id="{$id}"
+		{/if}
+	{/let}
+
+	<div {$attributes}>
+		{$unstable_content}
+
+		{call .input}
+			{param _handleOnBlur: $_handleOnBlur /}
+			{param _handleOnFocus: $_handleOnFocus /}
+			{param _handleOnInput: $_handleOnInput /}
+			{param _handleOnKeydown: $_handleOnKeydown /}
+			{param inputElementClasses: $inputElementClasses /}
+			{param inputName: $inputName /}
+			{param inputValue: $inputValue /}
+		{/call}
+
+		{call .dropdown}
+			{param _handleDataChange: $_handleDataChange /}
+			{param _handleDropdownItemClick: $_handleDropdownItemClick /}
+			{param contentRenderer: $contentRenderer /}
+			{param dataSource: $dataSource /}
+			{param filteredItems: $filteredItems /}
+			{param initialData: $initialData /}
+			{param requestOptions: $requestOptions /}
+			{param requestPolling: $requestPolling /}
+			{param requestRetries: $requestRetries /}
+			{param requestTimeout: $requestTimeout /}
+		{/call}
+	</div>
+{/template}
+
+/**
+ * This render the input tag.
+ */
+{template .input}
+	{@param? _handleOnBlur: any}
+	{@param? _handleOnFocus: any}
+	{@param? _handleOnInput: any}
+	{@param? _handleOnKeydown: any}
+	{@param? inputElementClasses: string}
+	{@param? inputName: string}
+	{@param? inputValue: string}
+
+	{let $attributes kind="attributes"}
+		class="
+			{if $inputElementClasses}
+				{$inputElementClasses}
+			{else}
+				form-control
+			{/if}
+		"
+
+		{if $inputName}
+			id="{$inputName}"
+			name="{$inputName}"
+		{/if}
+
+		onBlur="{$_handleOnBlur}"
+		onFocus="{$_handleOnFocus}"
+		onInput="{$_handleOnInput}"
+		onKeydown="{$_handleOnKeydown}"
+		ref="input"
+		type="text"
+
+		{if $inputValue}
+			value="{$inputValue}"
+		{else}
+			value=""
+		{/if}
+	{/let}
+
+	<input {$attributes} />
+{/template}
+
+/**
+ * This render the dropdown content.
+ */
+{template .dropdown}
+	{@param dataSource: string|[]|list<?>}
+	{@param? _handleDataChange: any}
+	{@param? _handleDropdownItemClick: any}
+	{@param? contentRenderer: string}
+	{@param? filteredItems: list<?>}
+	{@param? initialData: []|list<?>}
+	{@param? requestOptions: [
+		method: string,
+		mode: string,
+		cache: string,
+		credentials: string,
+		headers: [],
+		redirect: string,
+		referrer: string,
+		body: []
+	]}
+	{@param? requestPolling: number}
+	{@param? requestRetries: number}
+	{@param? requestTimeout: number}
+
+	{let $isVisible: $filteredItems and length($filteredItems) > 0 ? true : false /}
+
+	{let $classes kind="text"}
+		dropdown-menu
+		{if $isVisible}
+			{sp}show
+		{/if}
+	{/let}
+
+	{call ClayDataProvider.render}
+		{param content kind="html"}
+			<ul class="{$classes}" ref="dropdown">
```

Comment:
hey @jbalsas, yeah we need to have a `trigger` mandatory when using Dropdown (maybe we can leave this optional in the future...) and also the data is different, it's a bit difficult to leave it flexible, the data from the `deltemplates` of Dropdown are well crafted for use cases of it... I can try to add the dropdown afterwards and see how it would look today...